os/arch/arm/src/armv7-m/mpu.h

```diff
@@ -687,6 +704,35 @@ static inline void up_set_mpu_app_configuration(struct tcb_s *rtcb)
 #define up_set_mpu_app_configuration(x)
 #endif
 
+/****************************************************************************
+ * Name: up_set_mpu_stack_guard
+ *
+ * Description:
+ *   Configure task's stack guard region
+ *
+ ****************************************************************************/
+
+#if defined(CONFIG_MPU_STACK_OVERFLOW_PROTECTION)
+static inline void up_set_mpu_stack_guard(uint32_t region, struct tcb_s *rtcb)
+{
+	/* We update the MPU registers only if:
+	 * This is not a kernel thread AND
+	 * It has a non zero value of base address (This ensures valid MPU setting)
+	 */
+	int i = 3 * region;
+	if ((rtcb->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_KERNEL) {
```

Comment:
Is there any special reason not to support kernel thread?

Reply:
currently kernel is in privilege mode, so we need to disable it and MPU to kernel region.
There will be issues for accessing user space and other unknown issues as of now.